subprojects/ide-native/src/main/java/org/gradle/ide/xcode/plugins/XcodePlugin.java

```diff
@@ -0,0 +1,247 @@
+import org.gradle.api.tasks.Delete;
+import org.gradle.ide.xcode.XcodeExtension;
+import org.gradle.ide.xcode.internal.DefaultXcodeExtension;
+import org.gradle.ide.xcode.internal.DefaultXcodeProject;
+import org.gradle.ide.xcode.internal.XcodeScheme;
+import org.gradle.ide.xcode.internal.XcodeTarget;
+import org.gradle.ide.xcode.internal.xcodeproj.FileTypes;
+import org.gradle.ide.xcode.internal.xcodeproj.PBXTarget;
+import org.gradle.ide.xcode.tasks.GenerateSchemeFileTask;
+import org.gradle.ide.xcode.tasks.GenerateWorkspaceSettingsFileTask;
+import org.gradle.ide.xcode.tasks.GenerateXcodeProjectFileTask;
+import org.gradle.ide.xcode.tasks.GenerateXcodeWorkspaceFileTask;
+import org.gradle.language.swift.plugins.SwiftExecutablePlugin;
+import org.gradle.language.swift.plugins.SwiftModulePlugin;
+import org.gradle.plugins.ide.internal.IdePlugin;
+
+import java.io.File;
+import java.util.Set;
+
+/**
+ * A plugin for creating a XCode project for a gradle project.
+ */
+@Incubating
+public class XcodePlugin extends IdePlugin {
+    private static final Spec<Plugin> HAS_XCODE_PLUGIN = new Spec<Plugin>() {
+        @Override
+        public boolean isSatisfiedBy(Plugin element) {
+            return element instanceof XcodePlugin;
+        }
+    };
+    private static final Spec<Plugin> HAS_SUPPORTED_LANGUAGE_PLUGIN = new Spec<Plugin>() {
+        @Override
+        public boolean isSatisfiedBy(Plugin element) {
+            return element instanceof SwiftModulePlugin || element instanceof SwiftExecutablePlugin;
+        }
+    };
+    private DefaultXcodeExtension xcode;
+
+    @Override
+    protected String getLifecycleTaskName() {
+        return "xcode";
+    }
+
+    @Override
+    protected void onApply(final Project project) {
+        getLifecycleTask().setDescription("Generates XCode project files (pbxproj, xcworkspace, xcscheme)");
+        getCleanTask().setDescription("Cleans XCode project files (xcodeproj)");
+
+        xcode = project.getExtensions().create("xcode", DefaultXcodeExtension.class);
+        String filename = project.getPath();
+        if (isRoot(project)) {
+            filename = project.getName();
+        }
+        xcode.getProject().setLocationDir(project.file(filename + ".xcodeproj"));
+
+        configureForSwiftPlugin(project);
+
+        configureXcodeProject(project);
+        configureXcodeWorkspace(project);
+        configureXcodeCleanTask(project);
+    }
+
+    private void configureXcodeCleanTask(Project project) {
+        Delete cleanTask = project.getTasks().create("cleanXcodeProject", Delete.class);
+        cleanTask.delete(xcode.getProject().getLocationDir());
+        if (isRoot(project)) {
+            cleanTask.delete(toXcodeWorkspacePackageDir(project));
+        }
+        getCleanTask().dependsOn(cleanTask);
+    }
+
+    private void configureXcodeProject(final Project project) {
+        project.getPlugins().matching(HAS_SUPPORTED_LANGUAGE_PLUGIN).all(new Action<Plugin>() {
```

Comment:
I would generate an empty Xcode project if the Gradle project does not have any Swift plugins applied (or C/C++, etc plugins).